keps/sig-api-machinery/2896-openapi-v3/README.md

```diff
@@ -307,6 +307,10 @@ type OpenAPIv3 struct {
 }
 ```
 
+### Version Skew
+
+There is a potential version skew between old aggregated apiservers and a new kube-apiserver. All new aggregated apiservers will publish v3 which will be discovered and published by the aggregator for v3. Old aggregated apiservers will not publish v3, and the aggregator cannot discover the v3 schema for the corresponding aggregated apiservers. The v3 schema for the old apiservers will not be published by the aggregator until they are upgraded to support v3. Regardless of version skew, v2 aggregation will continue for all aggregated apiservers.
```

Comment:
is this really what we want? How realistic is it to convert v2->v3? Anything that stops us? Any downsides if we don't?

Reply:
Maybe we should slowly phase it out instead. We could mark the v2 as deprecated and warn the users it will be removed in future releases.

Reply:
I updated the section to provide a better rationale of not publishing a lossy v3 schema. +1 on eventually phasing out v2, but this would only be something we can start doing after v3 hits GA right?

Reply:
Can you clarify how a client using the openapi v3 would know that the information presented is incomplete and what a client consuming the openapi should do in such a situation?

Reply:
Since OpenAPIs are published per group/version and aggregated apiservers claim only a specific group/version, the g/v in question will provide a blank request on a retrieve the OpenAPI (We can also send a different header to indicate this). I don't believe we will have any partial v3 specs, they should either be fully v3 or not published at all.